### PR TITLE
Use repositories instead models

### PR DESCRIPTION
**Pull request by Matslom**

### CODE REVIEW

File: app/Database/Repositories/Eloquent/RoleRepository.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    protected $roleModel;
+
+
+    /**
+     * RoleRepository constructor.
+     * @param Role $roleModel The model to use for roles.
+     */
+    public function __construct(
+        Role $roleModel
+    ) {
```
Comment (euantorano) [inline app/Database/Repositories/Eloquent/RoleRepository.php]: Hi, this should probably be all on one line.

File: app/Database/Repositories/Eloquent/UserRepository.php [hunk 2]
```diff
@@ -136,4 +136,18 @@ public function findByUsername($username = '')
...
+	/**
+	 * Create a new user
+	 *
+	 * @param array $details Details about the user.
+	 *
+	 * @return User
+	 */
+	public function create(array $details = [])
+	{
+		$user = $this->userModel->create($details);
```
Comment (euantorano) [inline app/Database/Repositories/Eloquent/UserRepository.php]: Can be simplified to `return $this->userModel->create($details);`, removing the intermediate variable.

**Comment by euantorano** [comment]
Couple of small comments there, the style checks can be ignored for now.

**Comment by Matslom** [comment]
I wanted also add timestamps on creating user. After disabling them in model, there is no way to use with create method. 
Can we use something like this 
UserRepository.php
`$user = new $this->userModel;
$user->name = $details['name'];
$user->email = $details['email'];
...
$user->timestamps = true;
$user->save();
`
instead of this
`$this->userModel->create($details)`?

**Comment by euantorano** [comment]
That slight work, or you could set time stamps before create, then reset it.

**Comment by JoshHarmon** [comment]
My thought would be

```php
public function create(array $details)
{
    $user->create($details);
    $user->touch();
}
```

I would think that would work, but I have yet to test it.

**Comment by Matslom** [comment]
Unfortunately, not really
```
public function touch()
    {
        if (! $this->timestamps) {
            return false;
        }
```

**Comment by JoshHarmon** [comment]
Well... That's unfortunate. Try...

```php
public function create(array $details)
{
    // manually set timestamps after creating the model
    $user = $this->userModel->create($details);
    return $user->setCreatedAt($user->getFreshTimestamp())
                       ->setUpdatedAt($user->getFreshTimestamp());
}
```

**Comment by Matslom** [comment]
Yep, it works
```
$user->setCreatedAt($user->freshTimestamp())
			->setUpdatedAt($user->freshTimestamp())->save();
```

**Comment by JoshHarmon** [comment]
Also, it looks like you're using tabs. We're not using tabs anymore as of today, so if you could switch to 4 spaces per "tab", that would be great. Otherwise the code sniffer on Travis CI will freak out when we get this PR to go to it.

**Comment by Matslom** [comment]
:smile: I used 4 spaces in first commit. 
No problem, I will change it.

**Comment by JoshHarmon** [comment]
Sorry! We passed an internal RFC recently to change the standards to be plain PSR-2 with the prefix/suffix requirements added on!

File: app/Database/Repositories/Eloquent/UserRepository.php [hunk 3]
```diff
@@ -19,121 +19,140 @@
...
+     * Get all users active in the last x minutes
+     *
+     * @param int $minutes The number of minutes which are considered as "online time"
+     * @param string $orderBy
+     * @param string $orderDir
+<<<<<<< HEAD
+     * @param int $num     The number of users to return. Set to 0 to get all users
+=======
+     * @param int $num The number of users to return. Set to 0 to get all users
+>>>>>>> upstream/master
```
Comment (ATofighi) [inline app/Database/Repositories/Eloquent/UserRepository.php]: :-)

Comment (Matslom) [inline app/Database/Repositories/Eloquent/UserRepository.php]: I know. I created another PR https://github.com/mybb/mybb2/pull/229 Sorry about this.
This PR should should go to hell.

**Pull request closed without merging**